packages/scripts/src/monitor-oa/index.ts

```diff
@@ -0,0 +1,61 @@
+import { delay, Logger } from "@uma/financial-templates-lib";
+import { initMonitoringParams, startupLogLevel, waitNextBlockRange } from "./common";
+import { monitorAssertions } from "./MonitorAssertions";
+import { monitorDisputes } from "./MonitorDisputes";
+import { monitorSettlements } from "./MonitorSettlements";
+
+import type { BotModes } from "./common";
+
+const logger = Logger;
+
+async function main() {
+  const params = await initMonitoringParams(process.env);
+
+  logger[startupLogLevel(params)]({
+    at: "OAMonitor",
+    message: "Optimistic Asserter Monitor started 🔭",
+    botModes: params.botModes,
+  });
+
+  const cmds = {
+    assertionsEnabled: monitorAssertions,
+    disputesEnabled: monitorDisputes,
+    settlementsEnabled: monitorSettlements,
+  };
+
+  for (;;) {
+    // In case of non-zero polling delay waitNextBlockRange at the end of the loop could have returned the starting block
+    // to be greater than the ending block if there were no new blocks in the last polling delay. In this case we should
+    // wait for the next block range before running the commands.
+    if (params.blockRange.start > params.blockRange.end) {
+      params.blockRange = await waitNextBlockRange(params);
+      continue;
+    }
+
+    const runCmds = Object.entries(cmds)
+      .filter(([mode]) => params.botModes[mode as keyof BotModes])
+      .map(([, cmd]) => cmd(logger, params));
+
+    await Promise.all(runCmds);
```

Comment:
this pattern is great. each individual "Cmd" runs as it's own async call and then we run them all at once.